Tests/NiftyTests/diag_test.swift

```diff
@@ -1,45 +1,147 @@
-
 /***************************************************************************************************
  *  diag_test.swift
  *
  *  This file tests the diag function.
  *
- *  Author: Philip Erickson
- *  Creation Date: 22 Jan 2017
+ *  Author: Tor Rafsol Løseth
+ *  Creation Date: 07 Jul 2018
  *
  *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
  *  in compliance with the License. You may obtain a copy of the License at
  *
  *      http://www.apache.org/licenses/LICENSE-2.0
  *
- *  Unless required by applicable law or agreed to in writing, software distributed under the 
- *  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either 
- *  express or implied. See the License for the specific language governing permissions and 
+ *  Unless required by applicable law or agreed to in writing, software distributed under the
+ *  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing permissions and
  *  limitations under the License.
  *
- *  Copyright 2017 Philip Erickson
+ *  Copyright 2018 Tor Rafsol Løseth
  **************************************************************************************************/
 
 import XCTest
 @testable import Nifty
 
-class diag_test: XCTestCase 
+class diag_test: XCTestCase
 {
     #if os(Linux)
-    static var allTests: [XCTestCaseEntry] 
+    static var allTests: [XCTestCaseEntry]
     {
-        let tests = 
-        [
-            testCase([("testBasic", self.testBasic)]),
-        ]
-
+        let tests =
+            [
+                testCase([("testBasic", self.testBasic)]),
+                ]
+        
         return tests
     }
     #endif
-
-    func testBasic() 
-    {        
-        // TODO: fill me in
-        print("\n\t*** WARNING: Test unimplemented - \(#file)\n")
+    
+    func testBasic()
```

Comment:
Very nice! Some time let's talk about strategies for testing, whether the `testBasic` pattern I started with is really what we want to do or whether smaller test cases are better.